models/models.js

```diff
@@ -11,10 +11,29 @@ exports.selectAllTopics = () => {
     return db.query(queryStr)
 }
 
+exports.selectAllArticles = () => {
+    let queryStr = `SELECT articles.article_id, articles.author, title, topic, articles.created_at, articles.votes, article_img_url, COUNT (comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id;`
+
+    return db.query(queryStr)
+}
+
 exports.selectArticleById = (id) => {
     let queryStr = `SELECT * FROM articles`
 
     queryStr += ` WHERE article_id = ${id}`
 
     return db.query(queryStr)
+    .then(({ rows }) => {
+        if (rows.length === 0) {
+            return Promise.reject({status: 404, msg: 'article_id does not exist'})
+        } else {
+            return rows[0]
+        }
+    })
+    .catch((err) => {
```

Comment:
your catch should never be in your models, you should be passing any potential errors to a controller and catching them there